### PR TITLE
Add Re-try Handling Mechanism to Request Client

### PR DESCRIPTION
**Pull request by AJStangl**
Provided a mechanism for retrying requests. This should speed things up a little bit and reduce the over all amount of 4XX and 5XX responses that crash to processes. 

Slightly performance bump since we no longer handle sleeping explicitly.

### CODE REVIEW

File: model_finetuning/download_reddit_finetuning_data.py [hunk 1]
```diff
@@ -56,6 +63,30 @@ def clean_text(text):
 	return text
 
 
+def get_with_retry(link: str):
+	"""
+	Performs a request using a re-try mechanism. Throws if
+	"""
+	# This could be a configuration within a dataset ini maybe?
```
Comment (zacc) [inline model_finetuning/download_reddit_finetuning_data.py:70]: I think 99% of users won't need to adjust these parameters so I think it's better to 'kiss' and leave this all as it is.

File: model_finetuning/download_reddit_finetuning_data.py [hunk 2]
```diff
@@ -178,22 +209,23 @@ def main():
...
-
+					submission_response = get_with_retry(submission_search_link)
+					# TODO: Remove this but check that functionality remains the same.
 					if submission_response.status_code != 200:
 						if submission_attempt >= max_attempts:
 							print(f"Request error! Could not download submission date, status code {submission_response.status_code}")
 						elif verbose:
 							print(f"Request error! Status code {submission_response.status_code}, retrying (attempt {submission_attempt} of {max_attempts})")
-						time.sleep(0.4) # give it a little bit more time if the request was not successful
+							# No additional time should now be required.
```
Comment (zacc) [inline model_finetuning/download_reddit_finetuning_data.py:219]: I noticed that too; I think you can get rid of these `time.sleep` calls then

**Review by zacc — COMMENTED** [review]
Works really nice so far!

**Comment by AJStangl** [comment]
@zacc  This good on your end? I can start making additional improvements and adding things to the multi-branch. I think the speed here exposes some fun race conditions in the db writing but no one would notice given you need to re-start this thing every 2-6 hours.

**Comment by zacc** [comment]
@AJStangl , I cherry-picked the good commits from this pull request and put them directly onto the multibot-master branch.
Can you do your work on that branch from now on? As we want to use that one in the future, it has more features.

**Pull request closed without merging**